akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -128,6 +133,21 @@ private[remote] abstract class InboundCompression[T >: Null](
         activeVersion, nextTable.version, incomingTableVersion)
       OptionVal.None
     }
+  }
+
+  def confirmAdvertisement(tableVersion: Int): Unit = {
```

Comment:
ok, the more explicit way with ACK likely good anyway to avoid possible races we didn't think of yet.
Technically the first message using  new number I thought would be safe, did you specifically see that cause trouble?